lib/Server.js

```diff
@@ -625,6 +513,120 @@ class Server {
+          } catch (error) {
+            // ignore error
+          }
+
+          if (stats) {
+            // It is file
+            this.options.https[property] = fs.readFileSync(path.resolve(value));
+          } else {
+            this.options.https[property] = value;
+          }
+        }
+      }
+
+      let fakeCert;
+
+      if (!this.options.https.key || !this.options.https.cert) {
+        fakeCert = getCertificate(this.log);
+      }
+
+      this.options.https.key = this.options.https.key || fakeCert;
+      this.options.https.cert = this.options.https.cert || fakeCert;
+
+      // Only prevent HTTP/2 if http2 is explicitly set to false
+      const isHttp2 = this.options.http2 !== false;
+
+      // note that options.spdy never existed. The user was able
+      // to set options.https.spdy before, though it was not in the
+      // docs. Keep options.https.spdy if the user sets it for
+      // backwards compatibility, but log a deprecation warning.
+      if (this.options.https.spdy) {
+        // for backwards compatibility: if options.https.spdy was passed in before,
+        // it was not altered in any way
+        this.log.warn(
+          'Providing custom spdy server options is deprecated and will be removed in the next major version.'
+        );
+      } else {
+        // if the normal https server gets this option, it will not affect it.
+        this.options.https.spdy = {
+          protocols: ['h2', 'http/1.1'],
+        };
+      }
+
+      // `spdy` is effectively unmaintained, and as a consequence of an
+      // implementation that extensively relies on Node’s non-public APIs, broken
+      // on Node 10 and above. In those cases, only https will be used for now.
+      // Once express supports Node's built-in HTTP/2 support, migrating over to
+      // that should be the best way to go.
+      // The relevant issues are:
+      // - https://github.com/nodejs/node/issues/21665
+      // - https://github.com/webpack/webpack-dev-server/issues/1449
+      // - https://github.com/expressjs/express/issues/3388
+      if (semver.gte(process.version, '10.0.0') || !isHttp2) {
+        if (this.options.http2) {
+          // the user explicitly requested http2 but is not getting it because
+          // of the node version.
+          this.log.warn(
+            'HTTP/2 is currently unsupported for Node 10.0.0 and above, but will be supported once Express supports it'
+          );
+        }
+        this.listeningApp = https.createServer(this.options.https, this.app);
+      } else {
+        // The relevant issues are:
+        // https://github.com/spdy-http2/node-spdy/issues/350
+        // https://github.com/webpack/webpack-dev-server/issues/1592
+        // eslint-disable-next-line global-require
+        this.listeningApp = require('spdy').createServer(
+          this.options.https,
+          this.app
+        );
+      }
+    } else {
+      this.listeningApp = http.createServer(this.app);
+    }
+  }
```

Comment:
I think we should rename this method on `createServer`, also we can refactor this on two methods:
- setupHttps (logic for https)
- createServer (logic for `http.createServer`)